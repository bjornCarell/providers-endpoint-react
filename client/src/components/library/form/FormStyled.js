import styled from 'styled-components';

export const Form = styled.form.attrs(({ background, height, width }) => ({
  background: background || 'transparent',
  height: height || 'auto',
  width: width || '100%'
}))`
  background: ${props => props.background};
  border-radius: 5px;
  box-shadow: ${props =>
    props.boxShadow ? ({ theme }) => theme.animation.shadowAround : 'none'};
  /* display: flex;
  flex-direction: ${props => (props.flexRow ? 'row' : 'column')}; */
  height: ${props => props.height};
  /* justify-content: stretch; */
  margin-top: ${props =>
    props.noMargin ? '0rem' : ({ theme }) => theme.spacing.small};
  margin-left: ${props =>
    props.noMargin ? '0rem' : ({ theme }) => theme.spacing.small};
  padding: ${props =>
    props.noPadding ? '0rem' : ({ theme }) => theme.spacing.small};
  width: ${props => props.width};
`;
