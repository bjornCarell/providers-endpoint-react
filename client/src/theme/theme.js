export const theme = {
  color: {
    primary: '#004146',
    primaryLight: '#427783',
    primaryBright: '#E8F9F8',
    secondary: '#F89572',
    secondaryLight: '#FDDFD4',
    secondaryDark: '#FF8A61',
    secondaryBright: '#FFE8E5',
    white: '#FFFFFF',
    light: '#fcfcfc',
    black: '#262626',
    grey: '#F5F6F7'
  },
  spacing: {
    xss: '0.5rem',
    xs: '0.75rem',
    small: '1rem',
    medium: '2rem',
    large: '4rem',
    xl: '8rem',
    xxl: '16rem'
  },
  fontSize: {
    small: '1.2rem',
    medium: '1.6rem',
    large: '2.6rem',
    xl: '4.22rem',
    xxl: '6.85rem'
  },
  fontWeight: {
    small: 200,
    medium: 400,
    large: 600,
    xl: 800
  },
  animation: {
    shadowAround: '0px 0px 10px 0px rgba(0,0,0,0.1)',
    shadowAroundHover: '0px 0px 16px 0px rgba(0,0,0,0.13)',
    shadowBottom: '0px 4px 8px 0 rgba(0,0,0,0.1)',
    shadowBottomHover: '0px 8px 8px 0 rgba(0,0,0,0.12)',
    radiusS: '0.375rem',
    radiusM: '0.875rem',
    radiusL: '2rem',
    transition: 'all .25s cubic-bezier(0.4,0,0.2,1)'
  },
  zIndex: {
    sidebar: 4
  }
};
