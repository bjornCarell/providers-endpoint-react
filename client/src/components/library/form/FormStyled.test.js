import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import { theme } from '../../../theme/theme';
import 'jest-styled-components';
import { Form } from './FormStyled';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDom.render(<Form theme={theme} />, div);
});

it('renders default conditional styles', () => {
  const tree = renderer.create(<Form theme={theme}></Form>).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('background', 'transparent');
  expect(tree).toHaveStyleRule('height', 'auto');
  expect(tree).toHaveStyleRule('width', '100%');
  expect(tree).toHaveStyleRule('box-shadow', 'none');
  expect(tree).toHaveStyleRule('flex-direction', 'column');
  expect(tree).toHaveStyleRule('margin-top', `${theme.spacing.small}`);
  expect(tree).toHaveStyleRule('margin-left', `${theme.spacing.small}`);
  expect(tree).toHaveStyleRule('padding', `${theme.spacing.small}`);
});

it('renders conditional styles', () => {
  const tree = renderer
    .create(
      <Form
        theme={theme}
        background="#FFF"
        height="65rem"
        width="32.5rem"
        boxShadow
        flexRow
        noMargin
        noPadding
      ></Form>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('background', '#FFF');
  expect(tree).toHaveStyleRule('height', '65rem');
  expect(tree).toHaveStyleRule('width', '32.5rem');
  expect(tree).toHaveStyleRule('box-shadow', `${theme.animation.shadowAround}`);
  expect(tree).toHaveStyleRule('flex-direction', 'row');
  expect(tree).toHaveStyleRule('margin-top', '0rem');
  expect(tree).toHaveStyleRule('margin-left', '0rem');
  expect(tree).toHaveStyleRule('padding', '0rem');
});
