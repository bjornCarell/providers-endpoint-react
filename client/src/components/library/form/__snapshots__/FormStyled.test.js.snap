// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders conditional styles 1`] = `
.c0 {
  background: #FFF;
  border-radius: 5px;
  box-shadow: 0px 0px 10px 0px rgba(0,0,0,0.1);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 65rem;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  margin-top: 0rem;
  margin-left: 0rem;
  padding: 0rem;
  width: 32.5rem;
}

<form
  className="c0"
  height="65rem"
  width="32.5rem"
/>
`;

exports[`renders default conditional styles 1`] = `
.c0 {
  background: transparent;
  border-radius: 5px;
  box-shadow: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: auto;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  margin-top: 1rem;
  margin-left: 1rem;
  padding: 1rem;
  width: 100%;
}

<form
  className="c0"
  height="auto"
  width="100%"
/>
`;
